src/jit/codegenarm64.cpp

```diff
@@ -5579,6 +5592,210 @@ void CodeGen::genHWIntrinsicSimdUnaryOp(GenTreeHWIntrinsic* node)
     genProduceReg(node);
 }
 
+//------------------------------------------------------------------------
+// genHWIntrinsicSimdBinaryOverlapOp:
+//
+// Produce code for a GT_HWIntrinsic node with form SimdBinaryOverlapOp.
+//
+// Consumes two SIMD operands and produces a SIMD result.
+// First operand is both source and destination.
+//
+// Arguments:
+//    node - the GT_HWIntrinsic node
+//
+// Return Value:
+//    None.
+//
+void CodeGen::genHWIntrinsicSimdBinaryOverlapOp(GenTreeHWIntrinsic* node)
+{
+    GenTree*  op1       = node->gtGetOp1();
+    GenTree*  op2       = node->gtGetOp2();
+    var_types baseType  = node->gtSIMDBaseType;
+    regNumber targetReg = node->gtRegNum;
+
+    assert(targetReg != REG_NA);
+
+    genConsumeOperands(node);
+
+    regNumber op1Reg = op1->gtRegNum;
+    regNumber op2Reg = op2->gtRegNum;
+
+    assert(genIsValidFloatReg(op1Reg));
+    assert(genIsValidFloatReg(op2Reg));
+    assert(genIsValidFloatReg(targetReg));
+
+    instruction ins = getOpForHWIntrinsic(node, baseType);
+    assert(ins != INS_invalid);
+
+    bool     is16Byte = (node->gtSIMDSize > 8);
+    emitAttr attr     = is16Byte ? EA_16BYTE : EA_8BYTE;
+    insOpts  opt      = genGetSimdInsOpt(is16Byte, baseType);
```

Comment:
I would like to see the signature for `CodeGen::genGetSimdInsOpt(bool is16Byte, var_types elementType)` changed to `CodeGen::genGetSimdInsOpt(emitAttr size, var_types elementType) `and an asset added at the start of that method : `assert(isValidVectorDatasize(size);` 

All the callers to this method aleady use this unnecessary pattern to setup `is16Byte`
Also instead of `attr` we can use `size`, since we aren't going to have any GC flags for these instructions.

`emitAttr size     = is16Byte ? EA_16BYTE : EA_8BYTE;`

**I will do this for you on a follow up checkin.**

Reply:
@briansull Thanks.  May be better to handle in a different checkin as this appears to be used in other places as well.